src/test/app/PermissionedDomains_test.cpp

```diff
@@ -0,0 +1,557 @@
+            "alice7",
+            "alice8",
+            "alice9",
+            "alice10",
+            "alice11",
+            "alice12"};
+        std::unordered_map<std::string, Account> pubKey2Acc;
+        for (auto const& c : alice)
+            pubKey2Acc.emplace(c.human(), c);
+
+        for (int i = 0; i < accNum; ++i)
+            env.fund(XRP(1000), alice[i]);
+
+        // Create new from existing account with a single credential.
+        pdomain::Credentials const credentials1{{alice[2], "credential1"}};
+        {
+            env(pdomain::setTx(alice[0], credentials1));
+            BEAST_EXPECT(env.ownerCount(alice[0]) == 1);
+            auto tx = env.tx()->getJson(JsonOptions::none);
+            BEAST_EXPECT(tx[jss::TransactionType] == "PermissionedDomainSet");
+            BEAST_EXPECT(tx["Account"] == alice[0].human());
+            auto objects = pdomain::getObjects(alice[0], env);
+            auto domain = objects.begin()->first;
+            BEAST_EXPECT(domain.isNonZero());
+            auto object = objects.begin()->second;
+            BEAST_EXPECT(object["LedgerEntryType"] == "PermissionedDomain");
+            BEAST_EXPECT(object["Owner"] == alice[0].human());
+            BEAST_EXPECT(object["Sequence"] == tx["Sequence"]);
+            BEAST_EXPECT(
+                pdomain::credentialsFromJson(object, pubKey2Acc) ==
+                credentials1);
+        }
+
+        // Make longest possible CredentialType.
+        {
+            constexpr std::string_view longCredentialType =
+                "Cred0123456789012345678901234567890123456789012345678901234567"
+                "89";
+            static_assert(longCredentialType.size() == maxCredentialTypeLength);
+            pdomain::Credentials const longCredentials{
+                {alice[1], std::string(longCredentialType)}};
+
+            env(pdomain::setTx(alice[0], longCredentials));
+
+            // One account can create multiple domains
+            BEAST_EXPECT(env.ownerCount(alice[0]) == 2);
+
+            auto tx = env.tx()->getJson(JsonOptions::none);
+            BEAST_EXPECT(tx[jss::TransactionType] == "PermissionedDomainSet");
+            BEAST_EXPECT(tx["Account"] == alice[0].human());
+
+            bool findSeq = false;
+            for (auto const& [domain, object] :
+                 pdomain::getObjects(alice[0], env))
+            {
+                findSeq = object["Sequence"] == tx["Sequence"];
+                if (findSeq)
+                {
+                    BEAST_EXPECT(domain.isNonZero());
+                    BEAST_EXPECT(
+                        object["LedgerEntryType"] == "PermissionedDomain");
+                    BEAST_EXPECT(object["Owner"] == alice[0].human());
+                    BEAST_EXPECT(
+                        pdomain::credentialsFromJson(object, pubKey2Acc) ==
+                        longCredentials);
+                    break;
+                }
+            }
+            BEAST_EXPECT(findSeq);
```

Comment:
This can be simplified via using `ledger_entry`.

Reply:
No, because getObjects also check that there are no other objects for the account